src/p2p/net_node.inl

```diff
@@ -1957,6 +1983,52 @@ namespace nodetool
   }
   //-----------------------------------------------------------------------------------
   template<class t_payload_net_handler>
+  epee::net_utils::zone node_server<t_payload_net_handler>::send_txs(std::vector<cryptonote::blobdata> txs, const epee::net_utils::zone origin, const boost::uuids::uuid& source, const bool pad_txs)
+  {
+    namespace enet = epee::net_utils;
+
+    const auto send = [&txs, &source, pad_txs] (std::pair<const enet::zone, network_zone>& network)
+    {
+      if (network.second.m_notifier.send_txs(std::move(txs), source, pad_txs))
+        return network.first;
+      return enet::zone::invalid;
+    };
+
+    if (origin != enet::zone::invalid)
+      return send(*m_network_zones.begin()); // send all txs received via p2p over public network
+
+    if (m_network_zones.size() <= 2)
+      return send(*m_network_zones.rbegin()); // see static asserts below; sends over anonymity network iff enabled
+
+    /* These checks are to ensure that i2p is highest priority if multiple
+       zones are selected. Make sure to update logic if the values cannot be
+       in the same relative order. `m_network_zones` must be sorted map too. */
+    static_assert(std::is_same<std::underlying_type<enet::zone>::type, std::uint8_t>{}, "expected uint8_t zone");
+    static_assert(unsigned(enet::zone::invalid) == 0, "invalid expected to be 0");
+    static_assert(unsigned(enet::zone::public_) == 1, "public_ expected to be 1");
+    static_assert(unsigned(enet::zone::i2p) == 2, "i2p expected to be 1");
+    static_assert(unsigned(enet::zone::tor) == 3, "tor expected to be 2");
+
+    // check for anonymity networks with noise and connections
+    for (auto network = ++m_network_zones.begin(); network != m_network_zones.end(); ++network)
```

Comment:
This implicitely skips over the public zone, correct ? That seems fairly vulnerable to adding another public zone (if we, say, add GSM based connections somewhere, not that we'd do that but for the sake of example).

Reply:
Ok, so to rephrase so that it makes more sense to me - if someone added a public type network with an enum value of 4, then it would be treated incorrectly as an anonymity network. I will add a break in here if the zone "exceeds" tor since there I can't think of a `static_assert` that would work. Anyone adding an anonymity network should see that its not using it during testing and adjust.

Reply:
Maybe some templatey C++y thing like...

template<address_type t> struct is_public { static const bool value = false; }
template<> struct is_public<public_> { static const bool value = true; }

Or maybe even don't have a default so additions have to specify one.

Reply:
I added a check for when the zone is larger than the current expected max to break the processing loop. That trait style won't work because its a compile-time check, whereas this is a runtime value.